repos/fdbt-site/src/pages/api/reuseOperatorGroup.ts

```diff
@@ -12,34 +12,27 @@ import { updateSessionAttribute, getSessionAttribute } from '../../utils/session
 
 export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
     try {
-        const { reuseGroupChoice, premadeOperatorGroup } = req.body;
-        if (!reuseGroupChoice) {
+        const selectedOperatorGroup = Number(req.body.operatorGroupId);
+        const noc = getAndValidateNoc(req, res);
+
+        if (!selectedOperatorGroup) {
             updateSessionAttribute(req, REUSE_OPERATOR_GROUP_ATTRIBUTE, [
-                { errorMessage: 'Choose one of the options below', id: 'conditional-form-group' },
+                { errorMessage: 'Choose an operator group from the options below', id: 'operatorGroup' },
             ]);
             redirectTo(res, '/reuseOperatorGroup');
             return;
         }
 
-        if (reuseGroupChoice === 'No') {
-            updateSessionAttribute(req, REUSE_OPERATOR_GROUP_ATTRIBUTE, []);
-            redirectTo(res, '/searchOperators');
-            return;
-        }
+        const multipleOperators = await getOperatorGroupByNocAndId(selectedOperatorGroup, noc);
 
-        if (!premadeOperatorGroup) {
-            updateSessionAttribute(req, REUSE_OPERATOR_GROUP_ATTRIBUTE, [
-                { errorMessage: 'Choose a premade operator group from the options below', id: 'premadeOperatorGroup' },
-            ]);
-            redirectTo(res, '/reuseOperatorGroup');
-            return;
+        if (multipleOperators) {
```

Comment:
We are missing an 'else' on the end of this if statement. If the user modifies the form and changes their ID, and our DB call returns undefined, we need to be throwing an error instead of letting the code continue.

Reply:
Added the new case for the else and write a test case for the same.